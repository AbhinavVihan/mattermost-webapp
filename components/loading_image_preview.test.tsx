// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {screen} from '@testing-library/react';

import LoadingImagePreview from 'components/loading_image_preview';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('components/LoadingImagePreview', () => {
    test('should match snapshot', () => {
        const loading = 'Loading';
        const progress = 50;

        const wrapper = renderWithIntl(
            <LoadingImagePreview
                loading={loading}
                progress={progress}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(screen.getByTestId('loader-percent').textContent).toBe('Loading 50%');

        expect(wrapper).toMatchSnapshot();
    });

    test('should set the loading percent to 90', () => {
        const loading = 'Loading';
        const progress = 90;
        const wrapper = renderWithIntl(
            <LoadingImagePreview
                loading={loading}
                progress={progress}
            />,
        );

        expect(screen.getByTestId('loader-percent').textContent).toBe('Loading 90%');

        expect(wrapper).toMatchSnapshot();
    });
});
