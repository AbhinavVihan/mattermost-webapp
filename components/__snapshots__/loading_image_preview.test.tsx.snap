// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoadingImagePreview should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="view-image__loading"
      >
        <span
          class="LoadingSpinner"
          id="loadingSpinner"
        >
          <span
            class="fa fa-spinner fa-fw fa-pulse spinner"
            title="Loading Icon"
          />
        </span>
        <span
          class="loader-percent"
          data-testid="loader-percent"
        >
          Loading 50%
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="view-image__loading"
    >
      <span
        class="LoadingSpinner"
        id="loadingSpinner"
      >
        <span
          class="fa fa-spinner fa-fw fa-pulse spinner"
          title="Loading Icon"
        />
      </span>
      <span
        class="loader-percent"
        data-testid="loader-percent"
      >
        Loading 50%
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/LoadingImagePreview should match snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="view-image__loading"
      >
        <span
          class="LoadingSpinner"
          id="loadingSpinner"
        >
          <span
            class="fa fa-spinner fa-fw fa-pulse spinner"
            title="Loading Icon"
          />
        </span>
        <span
          class="loader-percent"
          data-testid="loader-percent"
        >
          Loading 50%
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="view-image__loading"
    >
      <span
        class="LoadingSpinner"
        id="loadingSpinner"
      >
        <span
          class="fa fa-spinner fa-fw fa-pulse spinner"
          title="Loading Icon"
        />
      </span>
      <span
        class="loader-percent"
        data-testid="loader-percent"
      >
        Loading 50%
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/LoadingImagePreview should set the loading percent to 90 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="view-image__loading"
      >
        <span
          class="LoadingSpinner"
          id="loadingSpinner"
        >
          <span
            class="fa fa-spinner fa-fw fa-pulse spinner"
            title="Loading Icon"
          />
        </span>
        <span
          class="loader-percent"
          data-testid="loader-percent"
        >
          Loading 90%
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="view-image__loading"
    >
      <span
        class="LoadingSpinner"
        id="loadingSpinner"
      >
        <span
          class="fa fa-spinner fa-fw fa-pulse spinner"
          title="Loading Icon"
        />
      </span>
      <span
        class="loader-percent"
        data-testid="loader-percent"
      >
        Loading 90%
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
